File: GameFramePro/Assets/Scripts/Core/Base/Module/NetWork/Download/Downloader/UnityWebRequest/ByteDataDownloadManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;
using UnityEngine.Networking;

namespace GameFramePro.NetWorkEx
{
    public class ByteDataDownloadManager : UnityWebRequestDownLoadManager
    {
        protected static ByteDataDownloadManager s_Instance = null;

        public static ByteDataDownloadManager S_Instance
        {
            get
            {
                if (s_Instance == null)
                    s_Instance = new ByteDataDownloadManager();
                return s_Instance;
            }
        } // 

        #region  各种类型获取数据的接口

        /// <summary>/// 获取一个数据/// </summary>
        public virtual UnityWebRequestDownloadTask GetDataFromUrl(string taskUrl, Action<UnityWebRequest, bool, string> callback, UnityTaskPriorityEnum priorityEnum = UnityTaskPriorityEnum.Normal)
        {
            if (string.IsNullOrEmpty(taskUrl))
            {
                Debug.LogError("TryAddToDownloadTaskLink Fail,Parameter is null of taskUrl");
                return null;
            }
            //TODO从下载的缓存中读取数据


            //将新的下载任务添加到缓存的下载队列中
            UnityWebRequestDownloadTask newDownLoadTask = GetDownloadTaskInstance(taskUrl, callback, priorityEnum);
            newDownLoadTask.ChangeDownloadState(TaskStateEum.Initialed);
            AllCacheDownLoadTasks.Add(newDownLoadTask);
            return newDownLoadTask;
        }


        protected UnityWebRequestDownloadTask GetDownloadTaskInstance(string taskUrl, Action<UnityWebRequest, bool, string> callback, UnityTaskPriorityEnum priorityEnum)
        {
            var newTask = mUnityWebRequestTaskPoolManager.GetItemFromPool();
            UnityWebRequest webRequest = new UnityWebRequest(taskUrl);
            webRequest.timeout = mTimeOut;
            webRequest.downloadHandler = new DownloadHandlerBuffer();
            newTask.InitialedDownloadTask(taskUrl, webRequest, priorityEnum, callback);
            return newTask;
        }

        #endregion
    }
}

File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/NetWorkEx/ByteDataDownloadManager.lua
---@class CS.GameFramePro.NetWorkEx.ByteDataDownloadManager : CS.GameFramePro.NetWorkEx.UnityWebRequestDownLoadManager
CS.GameFramePro.NetWorkEx.ByteDataDownloadManager = {}

---@property readonly CS.GameFramePro.NetWorkEx.ByteDataDownloadManager.S_Instance : CS.GameFramePro.NetWorkEx.ByteDataDownloadManager
CS.GameFramePro.NetWorkEx.ByteDataDownloadManager.S_Instance = nil

---@return CS.GameFramePro.NetWorkEx.ByteDataDownloadManager
function CS.GameFramePro.NetWorkEx.ByteDataDownloadManager()
end

---@param taskUrl : CS.System.String
---@param callback : CS.System.Action
---@param priorityEnum : CS.GameFramePro.NetWorkEx.UnityTaskPriorityEnum
function CS.GameFramePro.NetWorkEx.ByteDataDownloadManager:GetDataFromUrl(taskUrl, callback, priorityEnum)
end